### DISCUSSION TITLE
USB HOST [done]

### DISCUSSION THREAD

**Discussion by zircothc** [marcel-licence/esp32_basic_synth#14] [Ideas]
Hi. Nice work.

What do you think about use a MINI USB HOST SHIELD? 
And adding MIDI learn ;)
I used one of them with esp32 and its fun to be able to use a USB midi controller. I got akai MPC mini to launch samples (8 bit mozzi library) 
It comunicates by SPI. I dont know if adding that feature would break any other part of the code.
I'd love to have free time to test It. :(

**Comment by marcel-licence** [comment]
Hi,
thank you. Good idea. I also did some tests with a USB host shield. It works well and does also supports interfaces with multilpe MIDI connections. If one core is not used yet it would work without problems.
I never tested the mozzi library yet :-/
I think I could prepare something using a MIDI host. That may help a lot because new MIDI controllers often do not have a MIDI din connector. But I had problems with one controller, which seems not to be supported by the MIDI host.

**Reply by marcel-licence** [reply]
Hi,
now I managed to get it working in both directions: https://youtu.be/Mt3rT-SVZww
I will update this project soon and also create a standalone one for the USB stuff.
Best regards,
Marcel
